### PR TITLE
[4.0] Modify display and tip for Edit Article in Frontend

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
Since Email to a friend and Print article have been taken off from J4, we do not need the anymore the Cog and dropdown to choose among actions. We now need only the Edit Article when authorised to do so.
Use of `role="tooltip"]` instead of bootstrap tips.
Cassiopea `[role="tooltip"]` is modified for the look. Placement of tip specific to the article edit id.

Should also be OK for a11y. TO CHECK please as I'm not a specialist.

### Testing Instructions
3 situations:
1. Edit Article when article published // simplified tip
2. Edit Article when article is unpublished // simplified tip
3. Article is checked out by another user

In each case, hover the icon to see the tip.

patch, run NPM

### Actual result BEFORE applying this Pull Request
Screen Shot 2020-09-07 at 18 35 48

Screen Shot 2020-09-07 at 18 37 57

Screen Shot 2020-09-07 at 18 37 23

### Expected result AFTER applying this Pull Request
Screen Shot 2020-09-07 at 18 15 40

Screen Shot 2020-09-07 at 18 14 48

Screen Shot 2020-09-07 at 18 13 31

### Note
If this is OK, will do for Contact

### CODE REVIEW

File: administrator/components/com_content/src/Service/HTML/Icon.php [hunk 1]
```diff
@@ -145,58 +144,23 @@ public function edit($article, $params, $attribs = array(), $legacy = false)
...
-		$overlib .= '&lt;br&gt;';
-		$overlib .= $date;
-		$overlib .= '&lt;br&gt;';
-		$overlib .= Text::sprintf('COM_CONTENT_WRITTEN_BY', htmlspecialchars($author, ENT_COMPAT, 'UTF-8'));
-
-		$text = LayoutHelper::render('joomla.content.icons.edit', array('article' => $article, 'overlib' => $overlib, 'legacy' => $legacy));
+		$text = LayoutHelper::render('joomla.content.icons.edit', array('article' => $article, 'tooltip' => $tooltip, 'legacy' => $legacy));
 
-		$attribs['title']   = Text::_('JGLOBAL_EDIT_TITLE');
+		//$attribs['title']   = Text::_('JGLOBAL_EDIT_TITLE');
```
Comment (richard67) [inline administrator/components/com_content/src/Service/HTML/Icon.php]: Shall this commented out line be removed?

Comment (infograf768) [reply administrator/components/com_content/src/Service/HTML/Icon.php]: yep

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 741f9ca6be159bd9250ce3191be3710d8d09ec54<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

**Comment by brianteeman** [comment]
The accessibility changes you made are completely wrong
1. never add role=button to a link
2. incorrect usage of aria-live (and not needed)

**Comment by infograf768** [comment]
> The accessibility changes you made are completely wrong
 >   never add role=button to a link
 >   incorrect usage of aria-live (and not needed)

It was useless to set unsuccessful as I noted in the description `TO CHECK please as I'm not a specialist.`

Thank you for your comment. Getting rid of  these 2 aspects. Is there anything else?

**Comment by brianteeman** [comment]
I cannot replicate your screenshots - looks like you might be missing a class on the "tooltip" div

image

image

**Comment by brianteeman** [comment]
Also because the link is called edit and the tooltip begins with edit it is announced by a screenreader as
Edit Edit ....

**Comment by infograf768** [comment]
NPM ci is necessary.

Class is not missing, role tooltip is a class de facto
```
// set up hidden tooltip
[role="tooltip"]:not(.show) {
  z-index: $zindex-tooltip;
  display: none;
  max-width: 100%;
  padding: .5em;
  margin: .5em;
  color: $black;
  background: $white-offset;
  border: 1px solid $gray-600;
  border-radius: $border-radius;
  box-shadow: 0 0 .5rem rgba(0,0,0,.8);

  &[id^=editarticle-] {
    margin-inline-start: -10em;
  }

  [dir=ltr] & {
    text-align: left;
  }

  [dir=rtl] & {
    text-align: right;
  }
}
```

but it looks like I can't apply anymore this .diff in eclipse for the 3 layouts files
Checking that aspect now here.

Can you explain 
> Also because the link is called edit and the tooltip begins with edit it is announced by a screenreader as

Not sure what you mean by `the link is called edit`. Can you explain? And what it should be?

**Comment by infograf768** [comment]
I see now why this PR does ot apply anymore here..
https://github.com/joomla/joomla-cms/pull/30543 was merged

**Comment by infograf768** [comment]
The git apply issue solved itself here.. :)

**Comment by infograf768** [comment]
> Also because the link is called edit and the tooltip begins with edit it is announced by a screenreader as Edit Edit ....

Got it now. I don't think it is a big deal.

**Comment by brianteeman** [comment]
That would be a decision by the accessibility team

**Comment by infograf768** [comment]
Stefan made suggestions. Will modify stuff shortly.

**Comment by infograf768** [comment]
Modified with various improvements. aria-hidden true. Simplified tooltips.
Now we get:

Screen Shot 2020-09-08 at 17 28 10

Screen Shot 2020-09-08 at 17 24 42

Screen Shot 2020-09-08 at 17 26 56

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on f9fd8ed3c70e5736ff3caeb357275f62466b9b05<br /><br />I see the simplified tip in all three situations<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

File: layouts/joomla/content/icons/edit_lock.php [hunk 2]
```diff
@@ -9,11 +9,15 @@
...
 use Joomla\CMS\Language\Text;
 
+$article = $displayData['article'];
 $tooltip = $displayData['tooltip'];
 
+$aria_described = 'editarticle-' . (int) $article->id;
 ?>
-<span class="hasTooltip fas fa-lock fa-fw" title="<?php echo HTMLHelper::tooltipText($tooltip . '', 0); ?>"></span>
-<?php echo Text::_('JLIB_HTML_CHECKED_OUT'); ?>
+<span class="hasTooltip fas fa-lock"  aria-hidden="true"></span>
```
Comment (brianteeman) [inline layouts/joomla/content/icons/edit_lock.php]:
```suggestion
<span class="hasTooltip fas fa-lock" aria-hidden="true"></span>
```

remove space

File: templates/cassiopeia/scss/blocks/_form.scss [hunk 3]
```diff
@@ -74,10 +72,17 @@ td .form-control {
...
-  margin: .25em;
-  color: $white;
-  background: $black;
+  padding: .5em;
+  margin: .5em;
+  color: $black;
+  background: $white-offset;
+  border: 1px solid $gray-600;
+  border-radius: $border-radius;
+  box-shadow: 0 0 .5rem rgba(0,0,0,.8);
```
Comment (brianteeman) [inline templates/cassiopeia/scss/blocks/_form.scss]:
```suggestion
  box-shadow: 0 0 .5rem rgba(0, 0, 0, .8);
```

File: layouts/joomla/content/icons/edit.php [hunk 4]
```diff
@@ -25,7 +24,11 @@
 {
 	$icon = 'eye-slash';
 }
+$aria_described = 'editarticle-' . (int) $article->id;
 
 ?>
-<span class="hasTooltip fas fa-<?php echo $icon; ?> fa-fw" title="<?php echo HTMLHelper::tooltipText(Text::_('COM_CONTENT_EDIT_ITEM'), $overlib, 0, 0); ?>"></span>
-<?php echo Text::_('JGLOBAL_EDIT'); ?>
+<span class="fas fa-<?php echo $icon; ?>" aria-hidden="true"></span>
+ <?php echo Text::_('JGLOBAL_EDIT'); ?>
```
Comment (brianteeman) [inline layouts/joomla/content/icons/edit.php]:
```suggestion
	<?php echo Text::_('JGLOBAL_EDIT'); ?>
```

space to tab

**Comment by brianteeman** [comment]
Don't know why but the role=tooltip css is present in my unminified css file but NOR in the minified one

**Comment by chmst** [comment]
@brianteeman did you use the patchtester? I have the same effect, but think that this not related to the PR. The PR looks great I will test as soon as required the changes are comitted.

**Comment by brianteeman** [comment]
@chmst I did but I cant see how that would make a difference. If the code is present in the non-minified css then the patch has worked

**Comment by brianteeman** [comment]
found the problem - its not related to patchtester - creating a seperate issue for it

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on f9fd8ed3c70e5736ff3caeb357275f62466b9b05<br /><br />Marking as successful as the minification issue is unrelated to the actual pr<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

File: layouts/joomla/content/icons/edit_lock.php (hunk 2 above)
Comment (infograf768) [reply layouts/joomla/content/icons/edit_lock.php]: done

File: templates/cassiopeia/scss/blocks/_form.scss (hunk 3 above)
Comment (infograf768) [reply templates/cassiopeia/scss/blocks/_form.scss]: done

File: layouts/joomla/content/icons/edit.php (hunk 4 above)
Comment (infograf768) [reply layouts/joomla/content/icons/edit.php]: done

**Comment by infograf768** [comment]
> Don't know why but the role=tooltip css is present in my unminified css file but NOR in the minified one

Screen Shot 2020-09-09 at 07 39 21

**Comment by infograf768** [comment]
RTC as last changes were small cs corrections.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30588">issues.joomla.org/tracker/joomla-cms/30588</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

@wilsonge do we need to document the change that the helper function got removed in the upgrade guide?

**Comment by infograf768** [comment]
Will now make similar changes for the Edit Contact.

**Comment by wilsonge** [comment]
@laoneo I guess it can't hurt to document it although I suspect it's a touch overkill

**Pull request merged**